base/pkg/reqs.jl

```diff
@@ -91,14 +91,16 @@ function parse(lines::Vector{Line})
         if isa(line,Requirement)
             if !isempty(line.system)
                 applies = false
-                @windows_only applies |=  ("windows"  in line.system)
-                @unix_only    applies |=  ("unix"     in line.system)
-                @osx_only     applies |=  ("osx"      in line.system)
-                @linux_only   applies |=  ("linux"    in line.system)
-                @windows_only applies &= !("!windows" in line.system)
-                @unix_only    applies &= !("!unix"    in line.system)
-                @osx_only     applies &= !("!osx"     in line.system)
-                @linux_only   applies &= !("!linux"   in line.system)
+                if is_windows(); applies |=  ("windows"  in line.system); end
+                if is_unix();    applies |=  ("unix"     in line.system); end
+                if is_apple();   applies |=  ("osx"      in line.system); end
```

Comment:
Is someone actually working on adding iOS support?  Either way, the naming still doesn't seem quite right to me.  Windows, linux, and bsd are all operating system with multiple variants.  Apple is a company.

If we anticipate support for iOS soonish, then I would suggest `is_darwin`.  Otherwise, I still think `is_osx` is clearer and better parallel form.